dbms/src/Storages/Transaction/RegionTable.h

```diff
@@ -91,6 +97,21 @@ class RegionTable : private boost::noncopyable
     using TableMap = std::unordered_map<TableID, Table>;
     using RegionInfoMap = std::unordered_map<RegionID, TableID>;
 
+    // safe ts is maintained by check_leader RPC (https://github.com/tikv/tikv/blob/1ea26a2ac8761af356cc5c0825eb89a0b8fc9749/components/resolved_ts/src/advance.rs#L262),
+    // leader_safe_ts is the safe_ts in leader, leader will send <applied_index, safe_ts> to learner to advance safe_ts of learner, and TiFlash will record the safe_ts into safe_ts_map in check_leader RPC.
+    // self_safe_ts is the safe_ts in TiFlah learner. When TiFlash proxy receive <applied_index, safe_ts> from leader, TiFlash will update safe_ts_map when TiFlash has applied the raft log to applied_index.
+    struct SafeTsEntry
+    {
+        explicit SafeTsEntry(UInt64 leader_safe_ts, UInt64 self_safe_ts)
+            : leader_safe_ts(leader_safe_ts)
+            , self_safe_ts(self_safe_ts)
+        {}
+        std::atomic<UInt64> leader_safe_ts;
+        std::atomic<UInt64> self_safe_ts;
+    };
+    using SafeTsEntryPtr = std::unique_ptr<SafeTsEntry>;
```

Comment:
why here need to make it a unique_ptr?

Reply:
> why here need to make it a unique_ptr?

Because `atomic` can't be copy, so make it a unique_ptr for inserting `SafeTsEntry` to `safe_ts_map`.